lib/go-log/log_test.go

```diff
@@ -42,3 +42,15 @@ func TestUTC(t *testing.T) {
 		t.Errorf("expected UTC time, actual '" + actual + "'")
 	}
 }
+
+func TestNewlinesSingleLogLine(t *testing.T) {
+	buf := &bytes.Buffer{}
+	Init(nil, writeCloser{buf}, nil, nil, nil)
+	// Have to concatenate, because Go doesn't allow you to test multiple trailing newline behavior.
+	Errorln("foo\n\nbar\n" + "\n" + "\n")
```

Comment:
What does this mean? Surely Go doesn't just blatantly use different strings than the ones the programmer creates?

Reply:
That's truly crazy to me. I should be able to print as many blank lines as I want. And I'd bet that the compiler actually stores a single string constant rather than doing the concatenation at runtime as an optimization, so Go's basically refusing to allow you to do something it does itself.